packages/react/src/components/header/components/navigationLink/NavigationLink.tsx

```diff
@@ -0,0 +1,190 @@
+
+// import core base styles
+import 'hds-core';
+import styles from './NavigationLink.module.scss';
+import classNames from '../../../../utils/classNames';
+import { Link } from '../../../link';
+import { NavigationLinkDropdown, NavigationLinkInteraction, DropdownMenuPosition } from './navigationLinkDropdown';
+import { HeaderNavigationMenuContext } from '../headerNavigationMenu/HeaderNavigationMenuContext';
+import { DropdownDirection } from './types';
+
+export type NavigationLinkProps = Omit<
+  React.ComponentPropsWithoutRef<'a'>,
+  'target' | 'href' | 'onPointerEnterCapture' | 'onPointerLeaveCapture' | 'aria-label'
+> & {
+  /**
+   * Indicator for active link. This is used in HeaderNavigationMenu.
+   */
+  active?: boolean;
+  /**
+   * Additional class names to apply.
+   */
+  className?: string;
+  /**
+   * Set the direction for where the dropdown should appear. Use DropdownDirection.Dynamic for nested dropdowns as it sets the dropdown menu to the right, but if there's no space it'll put it to the left.
+   * @default DropdownDirection.Down;
+   */
+  dropdownDirection?: DropdownDirection;
+  /**
+   * Array of NavigationLink components to render in a dropdown. Can be used only inside navigation components.
+   */
+  dropdownLinks?: Array<React.ReactElement>;
+  /**
+   * Hypertext Reference of the link.
+   */
+  href: string;
+  /**
+   * Element index given by parent mapping.
+   * @internal
+   */
+  index?: number;
+  /**
+   * Label for link.
+   */
+  label: string;
+  /**
+   * Which sub navigation index is open.
+   * @internal
+   */
+  openSubNavIndex?: number;
+  /**
+   * Set which sub navigation index is open.
+   * @internal
+   */
+  setOpenSubNavIndex?: (val: number) => void;
+  /**
+   * Size of the link.
+   */
+  size?: 'S' | 'M' | 'L';
+};
+
+export const NavigationLink = ({
+  active,
+  className,
+  dropdownDirection = DropdownDirection.Down,
+  dropdownLinks,
+  href,
+  index,
+  label,
+  openSubNavIndex,
+  setOpenSubNavIndex,
+  size,
+  ...rest
+}: NavigationLinkProps) => {
+  const [isDropdownOpen, setDropdownOpen] = useState<boolean>(false);
+  const [dropdownOpenedBy, setDropdownOpenedBy] = useState<null | NavigationLinkInteraction>(null);
+  const [dynamicPosition, setDynamicPosition] = useState<null | DropdownMenuPosition>(null);
+  const { openMainNavIndex, setOpenMainNavIndex } = useContext(HeaderNavigationMenuContext);
+  const containerRef = useRef<HTMLSpanElement>(null);
+  const isSubNavLink = openSubNavIndex !== undefined && setOpenSubNavIndex !== undefined;
+
+  const handleDynamicMenuPosition = (val: boolean, e: React.MouseEvent) => {
+    // Null position when false so if user resizes browser, the calculation is done again
+    if (!val) setDynamicPosition(null);
+    else if (val && window === undefined) setDynamicPosition(DropdownMenuPosition.Right);
+    else {
+      const { clientX } = e;
+      // Pardon the magic number. It's the dropdown menu's width
```

Comment:
The clientX is the mouse position. Is it on purpose?
With keyboard this is 0. 
This might work in here: 
```typescript
else if(containerRef.current) {
      const { x:leftPosition, width } = containerRef.current.getBoundingClientRect();
      const rightPosition = leftPosition + width;
      const position = leftPosition > window.innerWidth - rightPosition ? DropdownMenuPosition.Left : DropdownMenuPosition.Right;
      setDynamicPosition(position);
    }
```

Reply:
Oh, you're right. I'll try this code out, thanks!!

Reply:
I think this good and works well enough. I'll update the demo with this and menu position changes.

Reply:
There might be a case where the menu would fit more perfectly on the right even tough the left side might have more space. But we can fix it after we get the accessibility audit results